crates/kona-providers/src/chain_provider.rs

```diff
@@ -0,0 +1,354 @@
+                                nonce: e.nonce,
+                                max_priority_fee_per_gas: e.max_priority_fee_per_gas,
+                                max_fee_per_gas: e.max_fee_per_gas,
+                                gas_limit: e.gas_limit as u128,
+                                to: e.to,
+                                value: e.value,
+                                input: e.input.clone(),
+                                access_list: alloy::eips::eip2930::AccessList(
+                                    e.access_list
+                                        .0
+                                        .clone()
+                                        .into_iter()
+                                        .map(|item| alloy::eips::eip2930::AccessListItem {
+                                            address: item.address,
+                                            storage_keys: item.storage_keys.clone(),
+                                        })
+                                        .collect(),
+                                ),
+                            };
+                            TxEnvelope::Eip1559(Signed::new_unchecked(eip_tx, sig, tx.hash))
+                        }
+                        Transaction::Eip4844(e) => {
+                            let eip_tx = TxEip4844 {
+                                chain_id: e.chain_id,
+                                nonce: e.nonce,
+                                max_fee_per_gas: e.max_fee_per_gas,
+                                max_priority_fee_per_gas: e.max_priority_fee_per_gas,
+                                max_fee_per_blob_gas: e.max_fee_per_blob_gas,
+                                blob_versioned_hashes: e.blob_versioned_hashes.clone(),
+                                gas_limit: e.gas_limit as u128,
+                                to: e.to,
+                                value: e.value,
+                                input: e.input.clone(),
+                                access_list: alloy::eips::eip2930::AccessList(
+                                    e.access_list
+                                        .0
+                                        .clone()
+                                        .into_iter()
+                                        .map(|item| alloy::eips::eip2930::AccessListItem {
+                                            address: item.address,
+                                            storage_keys: item.storage_keys.clone(),
+                                        })
+                                        .collect(),
+                                ),
+                            };
+                            TxEnvelope::Eip4844(Signed::new_unchecked(
+                                TxEip4844Variant::TxEip4844(eip_tx),
+                                sig,
+                                tx.hash,
+                            ))
+                        }
```

Comment:
is this all basically rpc tx -> TxEnvelope? Thinking alloy should have a conversion util for this

Reply:
Checking!

Reply:
These tx types seem to be from [reth::primitives](https://github.com/paradigmxyz/reth/blob/main/crates/primitives/src/transaction/mod.rs#L92).

Would a conversion util be in alloy to convert from reth primitive types?

Reply:
hmm, maybe instead in the rpc-types-compat crate in reth. Not a blocker for this PR, but an issue for these conversions would be nice if we don't already have them

Reply:
Agree alright, I'll get this in and put up an issue?

Reply:
Sounds good!